src/replication.c

```diff
@@ -2249,25 +2330,32 @@ void readSyncBulkPayload(connection *conn) {
         }
 
         zfree(server.repl_transfer_tmpfile);
-        server.repl_transfer_fd = -1;
+            server.repl_transfer_fd = -1;
         server.repl_transfer_tmpfile = NULL;
     }
 
     /* Final setup of the connected slave <- master link */
-    replicationCreateMasterClient(server.repl_transfer_s,rsi.repl_stream_db);
-    server.repl_state = REPL_STATE_CONNECTED;
+    if (isRdbConnectionSync(conn)) {
+        /* In case of fsync using rdb channel, the master client was already created for psync purposes
```

Comment:
let's avoid "fsync" as an alias for full-sync (i.e. it refers to fsync for files).
```suggestion
        /* In case of full-sync using rdb channel, the master client was already created for psync purposes
```